File: SqlOrganizeMy/DbMy.cs
using Microsoft.Extensions.Caching.Memory;
using MySql.Data.MySqlClient;
using SqlOrganize;

namespace SqlOrganizeMy
{
    /// <summary>
    /// Contenedor principal para mysql
    /// </summary>
    public class DbMy : Db
    {

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="config">
        /// Configuracion
        /// </param>
        /// <example>
        ///   connectionString = "server=127.0.0.1;uid=root;pwd=12345;database=test"
        /// </example>
        public DbMy(Config config, Schema schema, IMemoryCache? cache = null) : base(config, schema, cache)
        {
            /*
            prueba de conexion
            Las conexiones se realizan directamente cuando se requiere la eje-
            cucion de una consulta a la base de datos.
            Este codigo esta como referencia, se deja como ejemplo por si es
            necesario verificar la conexion                       
            var conn = new MySqlConnection();
            conn.ConnectionString = (string)config.connectionString;
            conn.Open();
            conn.Close();
            */
        }

        public override EntityPersist Persist()
        {
            return new EntityPersistMy(this);
        }

        public override EntitySql Sql(string entity_name)
        {
            return new EntitySqlMy(this, entity_name);
        }

        public override Query Query()
        {
            return new QueryMy(this);
        }

        public override Query Query(EntitySql sql)
        {
            return new QueryMy(this, sql);
        }

        public override Query Query(EntityPersist persist)
        {
            return new QueryMy(this, persist);
        }

        public override long GetMaxValue(string entityName, string fieldName)
        {
            return Sql(entityName).Select("CAST ( ISNULL( MAX($" + fieldName + "), 0) AS bigint)").Value<long>();
        }
    }
}

File: src/SqlOrganizeMy/Sql/DbMy.php
<?php

namespace SqlOrganizeMy\Sql;

use PDO;
use SqlOrganize\Sql\Db;
use SqlOrganize\Sql\Config;
use SqlOrganize\Sql\SelectQueries;
use SqlOrganize\Sql\ModifyQueries;


/**
 * Contenedor principal para SQL Server
 * 
 * SQL Server agrega espacios en blanco adicionales cuando se utiliza 
 * CONCAT y CONCAT_WS.
 */
class DbMy extends Db
{

    /**
     * Constructor
     * 
     * @example
     * $connectionString = "server=127.0.0.1;uid=root;pwd=12345;database=test"
     */
    public function __construct(Config $config)
    {
        parent::__construct($config);
    }

    protected function initPdo(){
        $this->pdo = new PDO("mysql:host=" . $this->config->host . ";dbname=" . $this->config->dbName, $this->config->user, $this->config->pass, [
                PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION,
                PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC
            ]);
        $this->pdo->exec("SET NAMES 'utf8mb3'");
    }


    public function CreateSelectQueries(): SelectQueries
    {
        return new SelectQueriesMy($this);
    }

    public function CreateModifyQueries(): ModifyQueries
    {
        return new ModifyQueriesMy($this);
    }

}